### PR TITLE
better Layer1 FCD examples

### PR DESCRIPTION
**Pull request by DZariusz**
this example follow phoenix v0.2.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,11 +48,21 @@ and you need to setup them into `.env`, see `example.env`.
...
 ```
 
-Open another console and run
+You should get result like this:
 
 ```shell
-npm run test
+  Umbrella First Class Data - Layer 1
+Deploying YourContract with Umbrella Chain: 0x10816ea91c8001b3a3890e8c597c5972b4e4abfc
+    getting latest numeric FCD for BTC-USD
```
Comment (avoloshko) [inline README.md:59]: I thought we were getting rid of FCD...

File: contracts/YourContract.sol [hunk 2]
```diff
@@ -19,16 +19,29 @@ contract YourContract {
...
+
+    function getLatestNumericFCD(bytes32 _key) public view returns (uint256) {
+        uint256 blockHeight = umbChain.getBlockHeight();
+        console.log("current block height:", blockHeight);
+
+        if (umbChain.getBlockVotersCount(blockHeight) > 0) {
+            return getNumericFCD(blockHeight, _key);
+        } else {
+            console.log("there is no consensus yet on this block:", blockHeight);
+            console.log("fetching previous");
```
Comment (avoloshko) [inline contracts/YourContract.sol:42]: what if there's no consensus in the previous block?
I think we can have a variable in Chain with the latest successful block.

**Review by federicorinaldi — APPROVED** [review]

**Comment by avoloshko** [comment]
Overall, I think we may want to make the contract a little bit applicable to some typical use-cases.
For example, we check the latest UmbChain price and make certain decisions such us sending token to someone.

Betting platform.
1) submitBet: ETH-USD > 1000 within 1 month
2) getWin(int betID) {
   Bet bet = bets[betID];
     if (chainUmb(bet.ticker) > bet.value && (block.timestamp - bet.time) < bet.duration) {
    umbToken.transfer(bet.address);
}
}

This is a very simple scenario, yet it can be used in in the real world.

**Review by avoloshko — CHANGES_REQUESTED** [review]
I think this particular contract
1) demonstrates some flaws in our design 
2) is to primitive

File: README.md (hunk 1 above)
Comment (DZariusz) [reply README.md:59]: I do not remember such a decision...

File: contracts/YourContract.sol (hunk 2 above)
Comment (DZariusz) [reply contracts/YourContract.sol:42]: there is **always** consensus in previous block (except of block 0 of course)

**Comment by DZariusz** [comment]
> Overall, I think we may want to make the contract a little bit more applicable to some typical use-cases.

the goal of this reference app is not to create product, but to show how to connect and use oracle. We need to keep it as simple as possible.

> demonstrates some flaws in our design

what flow?

> is to primitive

this is the whole point of this app - must be as simple as possible.

File: README.md (hunk 1 above)
Comment (avoloshko) [reply README.md:59]: we planned to rename it to
On-chain data, L2 data as far as I remember

File: contracts/YourContract.sol (hunk 2 above)
Comment (avoloshko) [reply contracts/YourContract.sol:42]: even for such a simple use-case the code looks unclear. Can we encapsulate this if statement inside Chain contract? 
So, we call something like getLastNumericFCD

File: contracts/YourContract.sol [hunk 3]
```diff
@@ -1,33 +1,68 @@
...
+    function fetchChainAddress() public {
+        umbChain = IChain(registry.getAddress("Chain"));
+        console.log("umbChain:");
+        console.logAddress(address(umbChain));
     }
 
-    function getLatestNumericFCD(bytes32 _key) public view returns (uint256) {
-        uint256 blockHeight = umbChain.getBlockHeight();
-        console.log("Fetching data from UMB Network for current block height", blockHeight);
+    function getBlockInfo(uint256 _blockHeight) public {
```
Comment (avoloshko) [inline contracts/YourContract.sol:34]: let's rename it to print block info

File: contracts/YourContract.sol [hunk 4]
```diff
@@ -1,33 +1,68 @@
...
-        uint256 value = umbChain.getSingleNumericData(blockHeight, _key);
+        console.log("root:");
+        console.logBytes32(blockData.root);
+        console.log("timestamp:", blockData.timestamp);
+        console.log("blockchain block number (anchor):", blockData.anchor);
+        console.log("minter:", blockData.minter);
+    }
+
+    function getLatestBlockInfo() public {
+        getBlockInfo(umbChain.getLatestBlockHeightWithData());
```
Comment (avoloshko) [inline contracts/YourContract.sol:47]: let's rename it to printLatestBlockInfo. It's not a getter

File: contracts/YourContract.sol [hunk 5]
```diff
@@ -1,33 +1,68 @@
...
 
     // get contract address from trusted source
-    constructor(address _umbChainAddress) public {
-        require(_umbChainAddress != address(0x0), "_umbChainAddress is empty");
+    constructor(address _registry) public {
+        require(_registry != address(0x0), "_registry is empty");
+
+        console.log("Deploying YourContract with Umbrella Registry:", _registry);
+        registry = IRegistry(_registry);
+        fetchChainAddress();
```
Comment (avoloshko) [inline contracts/YourContract.sol:25]: we do it here. Should we resolve it each time before people use it??

File: contracts/YourContract.sol [hunk 6]
```diff
@@ -1,33 +1,68 @@
...
+    }
 
-        console.log("Deploying YourContract with Umbrella Chain:", _umbChainAddress);
-        umbChain = IChain(_umbChainAddress);
+    function fetchChainAddress() public {
+        umbChain = IChain(registry.getAddress("Chain"));
```
Comment (avoloshko) [inline contracts/YourContract.sol:29]: should we call it every time before some business logic is executed?
- maybe we should let people update it. For example create a public/external method updateChainAddress()
- if we detach an old Chain contract, will it be destroyed?

**Comment by avoloshko** [comment]
👍

**Pull request merged**